server/src/routes/course/crossCheck/createDistribution.ts

```diff
@@ -7,18 +7,25 @@ import { CrossCheckStatus } from '../../../models/courseTask';
 import { CrossCheckDistributionService } from '../../../services/distribution';
 import { courseService, taskService } from '../../../services';
 import { setResponse } from '../../utils';
+import { isSubmissionDeadlinePassed } from './isSubmissionDeadlinePassed';
 
 const crossCheckDistributionService = new CrossCheckDistributionService();
 
 export const createDistribution = (_: ILogger) => async (ctx: Router.RouterContext) => {
   const { courseTaskId } = ctx.params;
 
   const courseTask = await taskService.getCourseTask(courseTaskId);
+
   if (courseTask == null) {
     setResponse(ctx, StatusCodes.BAD_REQUEST);
     return;
   }
 
+  if (!isSubmissionDeadlinePassed(courseTask.studentEndDate as string)) {
```

Comment:
why do you need that casting? you can efficiently pass both string and Date to Date constructor, in that way casting can be removed

Reply:
Done